ui/src/app/kvm/views/KVMDetails/KVMDetailsHeader/KVMDetailsHeader.tsx

```diff
@@ -3,34 +3,43 @@ import { useEffect } from "react";
 import { Spinner } from "@canonical/react-components";
 import { usePrevious } from "@canonical/react-components/dist/hooks";
 import { useDispatch, useSelector } from "react-redux";
-import { useParams } from "react-router";
 import { Link, useLocation } from "react-router-dom";
 
 import type { KVMSelectedAction, KVMSetSelectedAction } from "../KVMDetails";
 
 import SectionHeader from "app/base/components/SectionHeader";
-import type { RouteParams } from "app/base/types";
 import KVMActionFormWrapper from "app/kvm/components/KVMActionFormWrapper";
 import PodDetailsActionMenu from "app/kvm/components/PodDetailsActionMenu";
+import { getActionTitle as getPodActionTitle } from "app/kvm/utils";
+import { getActionTitle as getMachineActionTitle } from "app/machines/utils";
 import { actions as podActions } from "app/store/pod";
 import podSelectors from "app/store/pod/selectors";
+import type { Pod } from "app/store/pod/types";
 import { PodType } from "app/store/pod/types";
 import type { RootState } from "app/store/root/types";
 
 type Props = {
+  id: Pod["id"];
   selectedAction: KVMSelectedAction | null;
   setSelectedAction: KVMSetSelectedAction;
 };
 
+const getActionTitle = (selectedAction: KVMSelectedAction) => {
+  if (typeof selectedAction === "object" && "name" in selectedAction) {
+    return getMachineActionTitle(selectedAction.name);
+  }
+  return getPodActionTitle(selectedAction);
+};
+
 const KVMDetailsHeader = ({
+  id,
   selectedAction,
   setSelectedAction,
 }: Props): JSX.Element => {
   const dispatch = useDispatch();
   const location = useLocation();
-  const { id } = useParams<RouteParams>();
```

Comment:
This is already parsed from the parent component - it doesn't make any sense to parse it again here, so I've converted it to a prop